torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -179,26 +178,21 @@ bool nccl_use_nonblocking() {
   return nccl_use_nonblocking_;
 }
 
-int _parse_nccl_nonblocking_timeout() {
-  const char* val = getenv("TORCH_NCCL_NONBLOCKING_TIMEOUT");
-  int timeout = -1;
-  if (val) {
-    const std::string config(val);
-    timeout = std::stoi(config);
-    if (!nccl_use_nonblocking() && timeout > 0) {
-      TORCH_WARN(
-          "TORCH_NCCL_NONBLOCKING_TIMEOUT has no effect when TORCH_NCCL_USE_COMM_NONBLOCKING is false.");
-      timeout = -1;
+// Default value: 30 minutes
+int nccl_nonblocking_timeout() {
+  static int timeout = -2; // -2 means not initialized
+  if (timeout == -2) {
+    const char* val = getenv("TORCH_NCCL_NONBLOCKING_TIMEOUT");
```

Comment:
Use our new c10::utils::get_env?

Reply:
Yes, I am using it [here](https://github.com/pytorch/pytorch/pull/138527/files#diff-c4e79e1dee3eea72b89d4471b91de8a7dae284e27bd5c32ed3e5296a561e4ee9) in a PR stacked on top :)
And the current piece of code is being deleted in that PR.
Thanks for pointing that out!